src/lib/getNcuRc.ts

```diff
@@ -23,52 +24,47 @@ async function getNcuRc({
   const { default: chalkDefault, Chalk } = await import('chalk')
   const chalk = options?.color ? new Chalk({ level: 1 }) : chalkDefault
 
-  const rawResult = rcFile('ncurc', {
+  const rawResult = rcFile<RcOptions>('ncurc', {
     configFileName: configFileName || '.ncurc',
     defaultExtension: ['.json', '.yml', '.js'],
     cwd: configFilePath || (global ? os.homedir() : packageFile ? path.dirname(packageFile) : undefined),
   })
 
-  if (configFileName && !rawResult?.filePath) {
+  // ensure a file was found if expected
+  const filePath = rawResult?.filePath
+  if (configFileName && !filePath) {
     programError(options, `Config file ${configFileName} not found in ${configFilePath || process.cwd()}`)
   }
 
-  // @ts-expect-error -- rawResult.config is not typed thus TypeScript does not know that it has a $schema property
-  const { $schema: _, ...rawConfigWithoutSchema } = rawResult?.config || {}
-
-  const result = {
-    filePath: rawResult?.filePath,
-    // Prevent the cli tool from choking because of an unknown option "$schema"
-    config: rawConfigWithoutSchema,
-  }
+  // convert the config to valid options by removing $schema and parsing format
+  const { $schema: _, format, ...rawConfig } = rawResult?.config || {}
+  const config: Options = rawConfig
+  if (typeof format === 'string') config.format = format.split(',')
```

Comment:
Let's import and use [formatOption.parse](https://github.com/raineorshine/npm-check-updates/blob/8a5967a354e6b21ef7ae7180b929a93e0401cb77/src/cli-options.ts#L669) rather than duplicate the parsing logic. Having the same parsing logic in multiple places is more likely to introduce regressions in the future if one is changed and not the other.

Ideally we would generalize this to all options rather than only `format`, but I'm okay with it for the sake of limiting the scope of this PR. Thanks!

Reply:
> Ideally we would generalize this to all options

I considered that (and, in fact, an early version did exactly that), but I looked through all the options and not all the parsers would apply correctly. Some of the array ones would even break. In all other cases, the documented types match the actual, internal types. Format is the only weird one. 

Regardless, you're spot on about loading the parser function. I meant to do that after my sanity test and forgot. I'll get that changed.

Reply:
Very good points

Reply:
It's a good thing I went back! There was a pretty glaring bug in which it would drop any format option that _isn't_ a string. Clearly, I was getting tired by the time I finished this. 😭

Reply:
New commit should be better, but take a look to be sure.

Reply:
Good catch! Surprised that wasn't covered by the unit tests. 

I'll take a look soon.

Reply:
Me too! They seem so thorough, I got lazy. Embarrassing noob mistake on my part. :'(

Speaking of the tests, it's worth noting that they are impacted by the dev environment. When I first started, they were broken by my `~/.npmrc` having a default private registry. And they ask for my SSH key when testing against GitHub. Mocking those away would probably be a nightmare, but worth mentioning. Unfortunately, I don't have the bandwidth to contribute anything meaningful in that space.

Reply:
Yes, I agree. There are some live tests to intentionally cover the data that comes back from the npm registry, and some simply due to exceeding the current mocking capabilities of the project, but there is probably a better way to do this.

Reply:
If you have lots of contributors with this issue, you might consider putting un-mocked tests in a standardized container. But since most folks aren't going to have private registries configured, it's probably not worth the docker dependency. `mv ~/.npmrc ~/.npmrc.bak` was an easy workaround.